src/architecture/unixfs.md

```diff
@@ -0,0 +1,611 @@
+  before encoding, satisfying the above constraints.
+
+Implementations interpreting the `mtime` metadata in order to apply it within a
+non-IPFS target MUST observe the following:
+
+- If the target supports a distinction between `unspecified` and `0`/`1970-01-01T00:00:00Z`,
+  the distinction must be preserved within the target. For example, if no `mtime` structure
+  is available, a web gateway must **not** render a `Last-Modified:` header.
+- If the target requires an `mtime` ( e.g. a FUSE interface ) and no `mtime` is 
+  supplied OR the supplied `mtime` falls outside of the targets accepted range:
+  - When no `mtime` is specified or the resulting `UnixTime` is negative:
+    implementations must assume `0`/`1970-01-01T00:00:00Z` (note that such values
+    are not merely academic: e.g. the OpenVMS epoch is `1858-11-17T00:00:00Z`)
+  - When the resulting `UnixTime` is larger than the targets range ( e.g. 32bit
+    vs 64bit mismatch), implementations must assume the highest possible value
+    in the targets range. In most cases, this would be `2038-01-19T03:14:07Z`.
+
+## Paths
+
+Paths begin with a `<CID>/` or `/ipfs/<CID>/`, where `<CID>` is a [multibase]
+encoded [CID]. The CID encoding MUST NOT use a multibase alphabet that contains
+`/` (`0x2f`) unicode codepoints. However, CIDs may use a multibase encoding with
+a `/` in the alphabet if the encoded CID does not contain `/` once encoded.
+
+Everything following the CID is a collection of path components (some bytes)
+separated by `/` (`0x2F`). UnixFS paths read from left to right, and are
+inspired by POSIX paths.
+
+- Components MUST NOT contain `/` unicode codepoints because it would break
+  the path into two components.
+- Components SHOULD be UTF8 unicode.
+- Components are case-sensitive.
+
+### Escaping
+
+The `\` may be used to trigger an escape sequence. However, it is currently
+broken and inconsistent across implementations. Until we agree on a specification
+for this, you SHOULD NOT use any escape sequences and/or non-ASCII characters.
+
+### Relative Path Components
+
+Relative path components MUST be resolved before trying to work on the path:
+
+- `.` points to the current node and  MUST be removed.
+- `..` points to the parent node and MUST be removed left to right. When removing
+  a `..`, the path component on the left MUST also be removed. If there is no path
+  component on the left, you MUST error to avoid out-of-bounds
+  path resolution.
+
+### Restricted Names
+
+The following names SHOULD NOT be used:
+
+- The `.` string, as it represents the self node in POSIX pathing.
+- The `..` string, as it represents the parent node in POSIX pathing.
+- The empty string. <!--TODO: check that this is true-->
+- Any string containing a `NULL` (`0x00`) byte, as this is often used to signify string
+  terminations in some systems, such as C-compatible systems. Many unix
+  file systems do not accept this character in path components.
+
```

Comment:
Do we also want to restrict certain characters that may act as display control characters in some shells, such as backspace? While it would be somewhat of a departure from POSIX, it could help eliminate a class of security issues where portions of a path are effectively hidden when displayed in a shell.